lib/driver.js

```diff
@@ -86,6 +86,9 @@ class SelendroidDriver extends BaseDriver {
       await this.checkAppPresent();
       this.opts.systemPort = this.opts.selendroidPort || SYSTEM_PORT_RANGE[0];
       this.opts.adbPort = this.opts.adbPort || DEFAULT_ADB_PORT;
+      if (this.opts.reboot) {
```

Comment:
This work should be done in a `validateDesiredCaps` method on the Selendroid Driver (akin to the one on the Android driver: https://github.com/appium/appium-android-driver/blob/master/lib/driver.js#L396)